pkg/menu/menu.go

```diff
@@ -88,36 +88,45 @@ func processInput(introwords string, location int, wid int, ht int, isValid vali
 	ui.Render(warning)
 
 	// keep tracking all input from user
+	var fullText string
+
 	for {
 		k := readKey(uiEvents)
 		switch k {
 		case "<C-d>":
 			return input.Text, warning.Text, ExitRequest
 		case "<Enter>":
-			inputString, warningString, ok := isValid(input.Text)
+			inputString, warningString, ok := isValid(fullText)
 			if ok {
 				return inputString, warning.Text, nil
 			}
+			fullText = ""
 			input.Text = ""
 			warning.Text = warningString
 			ui.Render(input)
 			ui.Render(warning)
 		case "<Backspace>":
 			if len(input.Text) > 0 {
-				input.Text = input.Text[:len(input.Text)-1]
+				fullText = fullText[:len(fullText)-1]
+				start := max(0, len(fullText)-wid+3)
```

Comment:
Could you add a comment for future contributors explaining the numbers here? Like what is `wid` and why we add 3?